src/theory/quantifiers/ematching/candidate_generator.h

```diff
@@ -208,6 +212,26 @@ class CandidateGeneratorConsExpand : public CandidateGeneratorQE
   bool isLegalOpCandidate(Node n) override;
 };
 
+/**
+ * Candidate generator for selector applications, which considers both
+ * internal terms corresponding to correctly and incorrectly applied selectors.
+ */
+class CandidateGeneratorSelector : public CandidateGeneratorQE
+{
+ public:
+  CandidateGeneratorSelector(QuantifiersEngine* qe, Node mpat);
+  /** reset */
+  void reset(Node eqc) override;
+  /** get next candidate */
```

Comment:
It'd be good to describe how this method works in terms of correctly vs non-correctly applied selectors and so on.